core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.spec.ts

```diff
@@ -76,21 +82,156 @@ describe('DotEmaComponent', () => {
             const iframe = spectator.query(byTestId('iframe'));
             expect(iframe).toHaveAttribute('src', 'http://localhost:3000/page-one?language_id=2');
         });
+
+        it('should open a dialog when the iframe sends a postmessage with the edit-contenlet action', () => {
+            spectator.triggerEventHandler('select[data-testId="language_id"]', 'change', {
+                target: { name: 'language_id', value: '2' }
+            });
+
+            const dialog = spectator.query(byTestId('dialog'));
+
+            window.dispatchEvent(
+                new MessageEvent('message', {
+                    origin: 'http://localhost:3000',
+                    data: {
+                        action: 'edit-contentlet',
+                        payload: {
+                            inode: '123'
+                        }
+                    }
+                })
+            );
+
+            spectator.detectChanges();
+
+            expect(dialog.getAttribute('ng-reflect-visible')).toBe('true');
+        });
+
+        it('should not open a dialog when the iframe sends a postmessage with a different origin', () => {
+            spectator.triggerEventHandler('select[data-testId="language_id"]', 'change', {
+                target: { name: 'language_id', value: '2' }
+            });
+
+            const dialog = spectator.query(byTestId('dialog'));
+
+            window.dispatchEvent(
+                new MessageEvent('message', {
+                    origin: 'my.super.cool.website.xyz',
+                    data: {
+                        action: 'edit-contentlet',
+                        payload: {
+                            inode: '123'
+                        }
+                    }
+                })
+            );
+
+            spectator.detectChanges();
+
+            expect(dialog.getAttribute('ng-reflect-visible')).toBe('false');
+        });
+
+        it('should trigger onIframeLoad when the dialog is opened', () => {
+            jest.spyOn(spectator.component, 'onIframeLoad');
+            spectator.triggerEventHandler('select[data-testId="language_id"]', 'change', {
+                target: { name: 'language_id', value: '2' }
+            });
+
+            window.dispatchEvent(
+                new MessageEvent('message', {
+                    origin: 'http://localhost:3000',
+                    data: {
+                        action: 'edit-contentlet',
+                        payload: {
+                            inode: '123'
+                        }
+                    }
+                })
+            );
+
+            spectator.detectChanges();
+
+            const dialogIframe = spectator.debugElement.query(
+                By.css("[data-testId='dialog-iframe']")
+            );
+            spectator.triggerEventHandler(dialogIframe, 'load', {}); // There's no way we can load the iframe, because we are setting a real src and will not load
```

Comment:
But you can trigger here properties to assert what happens inside the `onIframeLoad`, no?

Reply:
The problem is that when you mock this:

`jest.spyOn(spectator.component, 'onIframeLoad');`

If I go to the component and remove all the body of `onIframeLoad` I think this test will not break.